File: examples/golang/atlas-basic.go
package golang

import (
	mdb "github.com/mongodb/awscdk-resources-mongodbatlas-go/awscdkresourcesmongodbatlas"
)

type CdkTestAppGoStackProps struct {
	awscdk.StackProps
}

func NewCdkTestAppGoStack(scope constructs.Construct, id string, props *CdkTestAppGoStackProps) awscdk.Stack {
	var sprops awscdk.StackProps
	if props != nil {
		sprops = props.StackProps
	}
	stack := awscdk.NewStack(scope, &id, &sprops)

	regionConfig := &mdb.AdvancedRegionConfig{
		AnalyticsSpecs: &mdb.Specs{
			EbsVolumeType: jsii.String("STANDARD"),
			InstanceSize:  jsii.String("M10"),
			NodeCount:     jsii.Number(1),
		},
		ElectableSpecs: &mdb.Specs{
			EbsVolumeType: jsii.String("STANDARD"),
			InstanceSize:  jsii.String("M10"),
			NodeCount:     jsii.Number(3),
		},
		Priority:   jsii.Number(7),
		RegionName: jsii.String("US_EAST_1"),
	}

	regionConfigs := &[]*mdb.AdvancedRegionConfig{regionConfig}
	replicationSpecs := &[]*mdb.AdvancedReplicationSpec{
		{
			AdvancedRegionConfigs: regionConfigs,
			NumShards:             jsii.Number(1),
		},
	}

	ipAccessListProps := mdb.IpAccessListProps{
		AccessList: &[]*mdb.AccessListDefinition{
			{
				Comment:   jsii.String("Open comment"),
				IpAddress: jsii.String("10.10.0.0/24"),
			},
		},
	}

	mdb.NewAtlasBasic(stack, jsii.String("AtlasBasic-go"), &mdb.AtlasBasicProps{
		ClusterProps: &mdb.ClusterProps{
			ReplicationSpecs: replicationSpecs,
		},
		ProjectProps: &mdb.ProjectProps{
			OrgId: jsii.String("63be77777777777705e"),
		},
		IpAccessListProps: &ipAccessListProps,
		Profile:           jsii.String("atlas-secret-profile-name"),
	})

	return stack
}

File: examples/l3-resources/atlas-basic.ts
// This example creates a project and a cluster in Atlas using the L1 resources.
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { AtlasBasic } from 'awscdk-resources-mongodbatlas';

interface AtlasStackProps {
  readonly orgId: string;
  readonly profile: string;
  readonly clusterName: string;
  readonly region: string;
  readonly ip: string;
}

export class CdkTestingStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const atlasProps = this.getContextProps();
    const atlasBasic = new AtlasBasic(this, 'AtlasBasic', {
        clusterProps: {
        name: atlasProps.clusterName,  
        replicationSpecs:   [
        {
            numShards: 1,
            advancedRegionConfigs: [
                {
                    analyticsSpecs: {
                        ebsVolumeType: "STANDARD",
                        instanceSize: "M10",
                        nodeCount: 1
                    },
                    electableSpecs: {
                        ebsVolumeType: "STANDARD",
                        instanceSize: "M10",
                        nodeCount: 3
                    },
                    priority:  7,
                    regionName: atlasProps.region,
                }]
        }]        
        },
        projectProps: {
          orgId: atlasProps.orgId,
        },
    
        ipAccessListProps: {
          accessList:[
            { ipAddress: atlasProps.ip, comment: 'My first IP address' }
          ]
        },
        profile: atlasProps.profile,
      });

  }

  getContextProps(): AtlasStackProps {
    const orgId = this.node.tryGetContext('orgId');
    if (!orgId){
      throw "No context value specified for orgId. Please specify via the cdk context."
    }

    const profile = this.node.tryGetContext('profile') ?? 'default';
    const clusterName = this.node.tryGetContext('clusterName') ?? 'test-cluster';
    const region = this.node.tryGetContext('region') ?? "US_EAST_1";
    const ip = this.node.tryGetContext('ip');
    if (!orgId){
      throw "No context value specified for ip. Please specify via the cdk context."
    }

    return {
      orgId,
      profile,
      clusterName,
      region,
      ip
    }
  }
}
